### DISCUSSION TITLE
Launch Option to Ignore the Autostart File

### DISCUSSION THREAD

**Discussion by ElliotLurie** [labwc/labwc#2308] [feature-requests]
Right now, there is no way to prevent `$XDG_CONFIG_HOME/labwc/autostart` from running when launching LabWC without changing the config directory and losing things like theme and key binding settings. This is very inconvenient when starting via a session manager like `xfce4-session` (tested with experimental Wayland support) that handles application auto-starting; you can get duplicate or conflicting applications/services like input methods. If a simple switch like `-a` were added, then startup scripts can just be modified to run `labwc -a` to prevent a conflict like this.

**Comment by ahesford** [comment]
I'm not convinced we should be adding arguments to ignore arbitrary parts of our documented startup sequence. Why autostart and not environment processing? What's the compelling case for either?

The simple and correct fix is to check for an environment variable in your autostart script and bail out if it's set. You can set this environment variable when launching labwc from your session manager.

**Reply by stefonarch** [reply]
There is also the -C option, to use different settings for different use cases if needed.

**Reply by ahesford** [reply]
Yes, and a few symlinks can ensure that the configuration is otherwise equivalent. There are many ways to solve this problem without complicating the command-line interface.

**Comment by Consolatis** [comment]
I have to agree to the points already raised. Depending on the way labwc is started there are various ways to archive this mixed use. From checking some env var in the autostart file as suggested above to making whatever starts the labwc session do some kind of `mv ~/.config/labwc/autostart{,.disabled}; labwc` and whatever starts labwc without session manager some kind of `mv ~/.config/labwc/autostart{.disabled,}; labwc`.

This whole thing is also only an issue if you switch between session manager and no session manager constantly in which case a separate config directory seems beneficial. Potentially including symlinks for the other files as suggested.
